backend/ltsa/LtsaService.cs

```diff
@@ -0,0 +1,286 @@
+using Microsoft.AspNetCore.WebUtilities;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Pims.Core.Exceptions;
+using Pims.Core.Http;
+using Pims.Core.Http.Models;
+using Pims.Ltsa.Configuration;
+using Pims.Ltsa.Extensions;
+using Pims.Ltsa.Models;
+using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+namespace Pims.Ltsa
+{
+    /// <summary>
+    /// LtsaService class, provides a service for integration with Ltsa API services.
+    /// </summary>
+    public class LtsaService : ILtsaService
+    {
+        #region Variables
+        private TokenModel _token = null;
+        private JsonSerializerOptions _jsonSerializerOptions = null;
+        private readonly JwtSecurityTokenHandler _tokenHandler;
+        private readonly ILogger<ILtsaService> _logger;
+        private readonly int MAX_RETRIES = 3;
+        #endregion
+        #region Properties
+        protected IHttpRequestClient Client { get; }
+        public LtsaOptions Options { get; }
+        #endregion
+
+        #region Constructors
+        /// <summary>
+        /// Creates a new instance of a LtsaService, initializes with specified arguments.
+        /// </summary>
+        /// <param name="options"></param>
+        /// <param name="client"></param>
+        public LtsaService(IOptions<LtsaOptions> options, IHttpRequestClient client, JwtSecurityTokenHandler tokenHandler, ILogger<ILtsaService> logger, IOptions<JsonSerializerOptions> serializerOptions)
+        {
+
+            this.Options = options.Value;
+            this.Client = client;
+            _tokenHandler = tokenHandler;
+            _logger = logger;
+            _jsonSerializerOptions = serializerOptions.Value;
+        }
+        #endregion
+
+        #region methods
+
+        /// <summary>
+        /// Send a request to the specified endpoint.
+        /// </summary>
+        /// <typeparam name="TR"></typeparam>
+        /// <param name="method"></param>
+        /// <returns></returns>
+        private async Task<TR> SendAsync<TR>(string url, HttpMethod method)
```

Comment:
Seeing the duplication we're incurring on multiple different api wrappers, wondering if it would be a good time to abstract this common code so that we don't have to copy+paste each time.